tests/kafkatest/services/kafka/kafka.py

```diff
@@ -164,8 +168,7 @@ def prop_file(self, node):
         prop_file = cfg.render()
         prop_file += self.render('kafka.properties', node=node, broker_id=self.idx(node),
                                  security_config=self.security_config,
-                                 interbroker_security_protocol=self.interbroker_security_protocol,
-                                 sasl_mechanism=self.sasl_mechanism)
+                                 interbroker_security_protocol=self.interbroker_security_protocol)
```

Comment:
Can we remove the use of `self.interbroker_security_protocol` here as well, and just get that info out of the `security_config`. Seems much better to rely on just the `SecurityConfig` instance.